atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformerJS.kt

```diff
@@ -0,0 +1,310 @@
+package kotlinx.atomicfu.transformer
+
+import org.mozilla.javascript.*
+import org.mozilla.javascript.ast.*
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.io.FileReader
+import java.lang.String.format
+
+class AtomicFUTransformerJS(
+    var inputDir: File,
+    var outputDir: File
+) {
+
+    private val p = Parser(CompilerEnvirons())
+    private val atomicConstructors = mutableSetOf<String>()
+
+    private operator fun File.div(child: String) =
```

Comment:
There is a bunch of copy-paste from `AtomicFUTransformer` class. Move the common parts into a separate utility class and/or extract some kind of `AtomicFUTransformerBase` with all the common code for both transformers.